### DISCUSSION TITLE
Idea: Social features

### DISCUSSION THREAD

**Discussion by bdbelchev** [MartsTech/Aeon#61] [Features] [answered]
We could implement features based on inter-user communication, for example gifting products to someone, having a friend list and possibly a timeline of your friends' recent purchases.

**Comment by MartsTech** [comment, marked as answer]
This is an interesting suggestions. The first stage of this idea will be the friends list. This will be made as a seperate services. Any user should be able to create an invite, share this invite by social media or email and accept the invite. Then every user must have a list of friends and be able to chat with them, using SignalR for example. Useres should be able to unfriend and block themselves.ft

As for the timeline of friends purchases, this will be a seperate service, that will save purchases as there is a completed order.

As for making gifts, the implementation will be for one user to make a cart and pay it with only needing to specify a billing address. For the user that is supposed to receive the gift, he will be notified in the chat that there is a gift for him and he will be able to complete the shipping by specifing a shipping address. Then the order will be saved for him only in the ordering service.

These are quite a few changes on both the frontend and the backend and I dont believe that they will be possible to implement right now. We may try some of them if there is time.
